packages/beacon-node/src/chain/archiveStore/archiveStore.ts

```diff
@@ -30,8 +32,17 @@ export class ArchiverStore {
 
     if (!opts.disableArchiveOnCheckpoint) return;
 
+    const mediator = new MediatorQueueObserver({
+      // As archive logic is splitted into separate observers, and each observer event is treated
+      // as it's own queue item. To keep matching with existing behavior multiplying old
+      // queue length with number of observers
+      maxQueueLength: PROCESS_FINALIZED_CHECKPOINT_QUEUE_LEN * 5,
```

Comment:
This should not be a "magic string" or "magic number" that gets input.  There should be a tie to the underlying queues that are getting merged and get updated via those values.  So updating this does not get overlooked if this file is modified (more or less observers)